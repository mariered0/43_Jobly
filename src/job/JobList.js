import React, { useState, useEffect, useContext } from "react";
import { Redirect } from "react-router-dom";
import JobCard from "./JobCard";
import SearchForm from "../common/SearchForm";
import UserContext from "../user/UserContext";
import Loading from "../common/Loading";
import JoblyApi from "../api/api";

const JobList = () => {
  const { token } = useContext(UserContext);

  const [jobs, setJobs] = useState([]);
  useEffect(() => {
    search();
  }, []);

  async function search(term) {
    if (!term) term = undefined;
    const getData = await JoblyApi.getJobs(term);
    console.log('jobs:', getData)
    setJobs(getData);
  }

  if(!jobs) return <Loading />;

  if (token) {
    return (
      <div className="container">
        <h1>JobList</h1>
        <SearchForm search={search} />

        {jobs.length ? (
          jobs.map((job) => (
            <JobCard
              key={job.id}
              title={job.title}
              salary={job.salary}
              equity={job.equity}
            />
          ))
        ) : (
          <p>No results found.</p>
        )}
      </div>
    );
  } else {
    return <Redirect to="/" />;
  }
};

export default JobList;
