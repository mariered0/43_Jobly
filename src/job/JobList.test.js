import React from "react";
import { render } from "@testing-library/react";
import JobList from "./JobList";
import JoblyApi from "../api/api";

import UserContext from "../user/UserContext";

const token = JoblyApi.token;

it("renders without crashing", () => {
  render(
        <UserContext.Provider value={{ token }}>
          <JobList />
        </UserContext.Provider>
  );
});

it("matches snapshot", () => {
  const { asFragment } = render(
        <UserContext.Provider value={{ token }}>
          <JobList />
        </UserContext.Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});