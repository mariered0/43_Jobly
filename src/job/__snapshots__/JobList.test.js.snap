// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <h1>
      JobList
    </h1>
    <div
      class="container"
    >
      <form
        class=""
      >
        <div
          class="row-cols-lg-auto g-3 row"
        >
          <div
            class="col-md-6 col-lg-8"
          >
            <input
              class="mb-3 form-control-lg form-control"
              id="term"
              name="term"
              placeholder="Enter search term..."
              type="text"
            />
          </div>
          <div
            class="col"
          >
            <button
              class="btn btn-secondary btn-lg"
              type="submit"
            >
              Submit
            </button>
          </div>
        </div>
      </form>
    </div>
    <p>
      No results found.
    </p>
  </div>
</DocumentFragment>
`;
